microprofile/grpc/metrics/src/main/java/io/helidon/microprofile/grpc/metrics/MetricsConfigurer.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.metrics;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.grpc.metrics.GrpcMetrics;
+import io.helidon.grpc.server.ServiceDescriptor;
+import io.helidon.microprofile.grpc.core.RpcMethod;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethod;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethodList;
+import io.helidon.microprofile.grpc.server.model.AnnotatedServiceConfigurer;
+import io.helidon.microprofile.grpc.server.model.ServiceModeller;
+import io.helidon.microprofile.metrics.MetricUtil;
+
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.annotation.Counted;
+import org.eclipse.microprofile.metrics.annotation.Metered;
+import org.eclipse.microprofile.metrics.annotation.Timed;
+
+import static io.helidon.microprofile.metrics.MetricUtil.getMetricName;
+import static io.helidon.microprofile.metrics.MetricUtil.registerMetric;
+
+/**
+ * A {@link AnnotatedServiceConfigurer} that adds a
+ * {@link io.helidon.grpc.metrics.GrpcMetrics gRPC metrics interceptor}
+ * to an annotated gRPC service.
+ * <p>
+ * Metric interceptors are only added to services where the method on the
+ * service class is annotated with a metric annotation. Any metric annotations
+ * on super classes or interfaces will be ignored.
+ */
+public class MetricsConfigurer
```

Comment:
Must this class be public? Seems like locally used.

Reply:
It is discovered in the gRPC MP server module using `ServiceLoader` so as far as I know has to be public.